### ISSUE TITLE
Got accessibility violation for Tooltip

### ISSUE THREAD

**Issue by lukewang2018** [IBM/carbon-components-react#1342]
## Detailed description

> Describe in detail the issue you're having.
When I use Dynamic Assessment Plugin to scan the accessibility issue, it's reporting below issue which is related to Tooltip. I guess it should add 'role' if showIcon is false. 

_The WAI-ARIA role(s) and/or attribute(s) identifier are not valid for the element identifier. See rule help for guidance._

See https://codesandbox.io/s/n9vkn3k870

> Is this issue related to a specific component?
Tooltip

> What did you expect to happen? What happened instead? What would you like to see changed?

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
   "carbon-components": "9.9.0",
    "carbon-components-react": "6.21.4",
    "carbon-icons": "7.0.7",
    "react": "16.4.0",
    "react-dom": "16.4.0",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

1. Access https://codesandbox.io/s/n9vkn3k870
2. Scan the page with DAP plugin
3. Check violation as the snapshot.

> Please create a reduced test case in code sandbox
> https://codesandbox.io/s/n9vkn3k870

## Additional information

* Screenshots or code
* Notes
![snip20180920_4](https://user-images.githubusercontent.com/40848250/45810403-28ad6480-bcfd-11e8-95b5-26a2056fdf14.png)
![snip20180920_1](https://user-images.githubusercontent.com/40848250/45810476-38c54400-bcfd-11e8-8e65-d699a5a2f8d9.png)

https://www.w3.org/TR/html-aria/#docconformance

**Comment by joshblack** [comment]
cc @IBM/carbon-designers do we support this tooltip style still? Having a hard time finding it in: http://www.carbondesignsystem.com/components/tooltip/code

**Comment by asudoh** [comment]
@joshblack We were meant to remove hover-over tooltip, however, a new use case has come along after then, which is, icon-based form validation.

**Comment by dakahn** [comment]
@asudoh @joshblack I'm going to close this one up since it's a use case we don't officially support. If that change though I'm happy to re-open!
